src/lib.rs

```diff
@@ -735,6 +737,27 @@ pub enum BinaryOperator {
     ShiftRight,
 }
 
+/// Function on an atomic value.
+///
+/// Note: these do not include load/store, which use the existing
+/// [`Expression::Load`] and [`Statement::Store`].
```

Comment:
If WGSL's `atomicLoad`, `atomicStore`, etc. ever expand to take other parameters (say, a memory ordering, like atomic loads in most other languages), then for Naga to handle them as ordinary `Load` / `Store` expressions/statements is going to cause an awkward change. I think these should be special `AtomicFunction` cases.

Reply:
Let's cross that bridge later. It's more difficult than it seems at first, because of how GLSL and SPIR-V treat them. I.e. it can be an ordinary value for some time (using regular load/store), and then we'll see the atomic op on it. Just keeping the existing Load/Store expressions makes this much more sane to handle.